engine/src/exec-interface-image.cpp

```diff
@@ -344,14 +352,39 @@ void MCImage::GetText(MCExecContext& ctxt, MCDataRef& r_text)
 		}
 
 		if (MCDataCreateWithBytes((const byte_t *)t_data, t_size, r_text))
-			return;
+        {
+            MCactiveimage = this;
+            return true;
+        }
 	}
 	
-	ctxt . Throw();
+	return false;
+}
+
+void MCImage::GetText(MCExecContext& ctxt, MCDataRef& r_text)
```

Comment:
Just an addendum - you don't need to recompress() on the image if it has been cloned from 'this' - as the copy constructor implicitly compresses the imagedata anyway. i.e. recompress() only needs to be called if the rep is *not* locked.